stats/stats-server/src/update_groups.rs

```diff
@@ -0,0 +1,87 @@
+use stats::{
+    construct_update_group,
+    counters::{
+        AverageBlockTime, CompletedTxns, LastNewContracts, LastNewVerifiedContracts, TotalAccounts,
+        TotalAddresses, TotalBlocks, TotalContracts, TotalNativeCoinHolders,
+        TotalNativeCoinTransfers, TotalTokens, TotalTxns, TotalVerifiedContracts,
+    },
+    lines::{
+        AccountsGrowth, ActiveAccounts, AverageBlockRewards, AverageBlockSize, AverageGasLimit,
+        AverageGasPrice, AverageTxnFee, ContractsGrowth, GasUsedGrowth, NativeCoinHoldersGrowth,
+        NativeCoinSupply, NewAccounts, NewBlocks, NewContracts, NewNativeCoinHolders,
+        NewNativeCoinTransfers, NewTxns, NewVerifiedContracts, TxnsFee, TxnsGrowth,
+        TxnsSuccessRate, VerifiedContractsGrowth,
+    },
+};
+
+macro_rules! singleton_groups {
+    ($($chart: ident),+ $(,)?) => {
+        $(
+            ::paste::paste!(
+                construct_update_group!([< $chart Group >] {
+                    charts: [$chart]
+                });
+            );
+        )+
+    };
+}
+
+singleton_groups!(
+    ActiveAccounts,
+    AverageBlockRewards,
+    AverageBlockSize,
+    AverageGasLimit,
+    AverageGasPrice,
+    AverageTxnFee,
+    GasUsedGrowth,
+    NativeCoinSupply,
+    NewBlocks,
+    TxnsFee,
+    TxnsSuccessRate,
+    AverageBlockTime,
+    CompletedTxns,
+    TotalAddresses,
+    TotalBlocks,
+    TotalTokens,
+);
+
+construct_update_group!(NewAccountsGroup {
+    charts: [NewAccounts, AccountsGrowth, TotalAccounts]
+});
+
+construct_update_group!(NewContractsGroup {
+    charts: [
+        NewContracts,
+        ContractsGrowth,
+        // currently can be in a separate group but after #845
+        // it's expected to join the group. placing it here to avoid
+        // updating config after the fix (todo: remove (#845))
+        TotalContracts,
```

Comment:
I didn't get it. 

Total Contracts should be in separated groups, right?
`stats` cannot recalculate charts after indexing of internal transactions, so we wrote separate SQL for total_contract. But now it looks like you're merging it again

Reply:
We still have the separate SQL, `TotalContracts` being in this group only means that its update will be triggered simultaneously to other charts in the group.

Why I added it:
Let's say `TotalContracts` is in it's own group. Now there's an additional group in the config. Once (if) #845 is fixed, `TotalContracts` will have to be moved to `NewContractsGroup` anyway. However, this means that the configs have to be updated as well.
Therefore, by adding it to the group in advance we won't need to make additional config changes.

Reply:
Feel free to reopen the thread; I'll close it for now